internal/admission/validation/gateway/httproute.go

```diff
@@ -97,22 +98,69 @@ func validateHTTPRouteListener(listener *gatewayapi.Listener) error {
 // HTTPRoute implementation and validates that the provided object is not using
 // any of those unsupported features.
 func validateHTTPRouteFeatures(httproute *gatewayapi.HTTPRoute, translatorFeatures translator.FeatureFlags) error {
-	for _, rule := range httproute.Spec.Rules {
-		for _, match := range rule.Matches {
-			// We support query parameters matching rules only with expression router.
-			if len(match.QueryParams) != 0 {
-				if !translatorFeatures.ExpressionRoutes {
-					return fmt.Errorf("queryparam matching is supported with expression router only")
-				}
+	unsupportedHTTPRouteFilterTypes := []gatewayapi.HTTPRouteFilterType{
+		gatewayapi.HTTPRouteFilterURLRewrite,
+		gatewayapi.HTTPRouteFilterRequestMirror,
+	}
+	unsupportedFilterMap := lo.SliceToMap(unsupportedHTTPRouteFilterTypes, func(t gatewayapi.HTTPRouteFilterType) (gatewayapi.HTTPRouteFilterType, struct{}) {
+		return t, struct{}{}
+	})
```

Comment:
Can't we just declare a `map[gatewayapi.HTTPRouteFilterType]struct{}` upfront instead of declaring a slice and converting it into a map? How is this an improvement?

Reply:
done.